services/order/handler.go

```diff
@@ -175,12 +173,45 @@ func (o *orderHandler) handleConfirmDeliveryStatus(w http.ResponseWriter, r *htt
 	if err!= nil {
         utils.WriteError(w, http.StatusInternalServerError, "Failed to update order status")
         return
+    }
+	// get orderby the id
+	orderResponse, err := o.store.GetOrderByID(orderID)
+	if err != nil {
+		utils.WriteError(w, http.StatusInternalServerError)
+		return
+	}
+	riderID := orderResponse.RiderID
+	rider, err := o.riderStore.GetRiderByID(riderID)
+	if err!= nil {
+        utils.WriteError(w, http.StatusInternalServerError, "Failed to get rider")
+        return
+    }
+	riderUserID := rider.UserID
+	chargeAmount := orderResponse.Charge
+	riderUser, err := o.userStore.GetUserByID(int(riderUserID))
+	if err!= nil {
+        utils.WriteError(w, http.StatusInternalServerError, "Failed to get rider user")
+        return
     }
 	// TODO: add charge amount to rider wallet
+	err = riderUser.Credit(o.db, chargeAmount)
+    if err!= nil {
+        utils.WriteError(w, http.StatusInternalServerError, "Failed to credit rider wallet")
+        return
+    }
+    // TODO: update rider successful ride by 1
+	err = rider.UpdateSuccessfulRides(o.db)
+	if err!= nil {
+        log.Println("Error updating Rider Successful rides")
+    }
+
+    // TODO: add email notification
+	message := fmt.Sprintf("Your order delivery has been successful confirmed. ₦%.1f has been added to your wallet", chargeAmount)
+	subject := "Order Delivery Notification"
 
-	// TODO: add email notification
+	go utils.SendMail(riderUser.Email, subject, riderUser.UserName, message)
```

Comment:
Yeah, I'm aware but choose not to handle the error here, cause it been handled and logged from utils function, if there's an error it will be logged continue the routine.

Reply:
Thank you for the look out